src/components/Game/GameArea.js

```diff
@@ -0,0 +1,138 @@
+import { useState } from 'react';
+
+import ActionButtons from 'components/Game/ActionButtons';
+import History from 'components/Game/History';
+import HealthMeter from "components/Game/HealthMeter";
+
+const PLAYER_ATTACK_MIN_PERCENTAGE = 1;
+const PLAYER_ATTACK_MAX_PERCENTAGE = 10;
+const MONSTER_ATTACK_MIN_PERCENTAGE = 1;
+const MONSTER_ATTACK_MAX_PERCENTAGE = 10;
+const SPECIAL_ATTACK_MIN_PERCENTAGE = 11;
+const SPECIAL_ATTACK_MAX_PERCENTAGE = 20;
+const SPECIAL_ATTACK_MIN_HEALTH = 90;
+const PLAYER_HEAL_PERCENTAGE = 10;
+const GAME_OVER_HEALTH = 0;
+
+const GameArea = (() => {
+  const gameState = {
+    player: 100,
+    enemy: 100,
+    started: false
+  };
+
+  const historyState = {
+    movesLog: []
+  };
+  
+  const randomNumber = ((min, max) => {
+    return Math.floor(Math.random() * (max - min)) + min;
+  });
+
+  const historyMessage = ((actionDoneBy, damage, actionType="attack") => {
+    if(actionType == "heal") {
+      return `Player heals for ${damage}`;
+    }
+    if(actionDoneBy == "player") {
+      return `Player hits monster for ${damage}`;
+    }
+    return `Monster hits player for ${damage}`;
+  });
+
+  const [currentGameState, updateGameState] = useState(gameState);
```

Comment:
```suggestion
  const [currentGameState, updateGameState] = useState(gameState);
```
1. We can drop the `game` prefix.
2. `current` is not required

Reply:
We need game prefix to differentiate between gametstate and historystate
current is required to differentiate between default and current state

Reply:
Prefix and `Current` is not required as -
1. You're writing inside Game component, so adding game prefix is redundant. 
2. `Current` is redundant information. If you need to differentiate between current and default, you can always name your default state as DEFAULT_GAME_STATE, and the current game state as maybe, gameState.
3. Change updateGameState to setGameState - as update has an inference of merging, which is not the case here. It sets the value of state.